### ISSUE TITLE
Error when trying to approve ERC721 token - Error: Unable to get token total supply

### ISSUE THREAD

**Issue by matias-gmg** [block-wallet/extension#438]
**Describe the bug**
I am getting the following error when trying to submit an ERC721 approve transaction using `web3.js`:

```
Error: Unable to get token total supply
    at i (blankProvider.js:1:3529)
    at Object.internal (blankProvider.js:1:4136)
    at t.validateError (blankProvider.js:1:42591)
    at h.handleResponse (blankProvider.js:1:33534)
    at blankProvider.js:1:32167
```

The code I am executing is pretty straightforward and it works in MetaMask:

```
      const erc721 = new web3.eth.Contract(ERC721ABI.abi, selectedToken[0]);
      try {
        await erc721.methods.approve(approveTo, selectedToken[2]).send({
          from: account,
        });
      } catch (err) {
        console.log(err);
      }
```

Why do I need to know the total supply of an ERC721 for approval? Maybe this is the problem as this ERC721 does not have `totalSupply` method in its interface (it does not inherits `ERC721Enumerable`).  

**To Reproduce**
Execute the following code in a `React` application:

```
      const erc721 = new web3.eth.Contract(ERC721ABI.abi, selectedToken[0]);
      try {
        await erc721.methods.approve(approveTo, selectedToken[2]).send({
          from: account,
        });
      } catch (err) {
        console.log(err);
      }
```

**Expected behavior**
I am expecting the wallet to popup the dialog to submit the approval transaction but I do not get it.

**Screenshots**
![Selection_307](https://user-images.githubusercontent.com/124641980/232971795-3a65c421-3c96-4623-ae5c-c60d967d3d74.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.5 LTS (64-bit)
 - Browser: Brave
 - Version 1.49.120 Chromium 111.0.5563.64 (Official Build) (64-bit)

**Additional context**

**Comment by matias-gmg** [comment]
Hi, I am having this error now:

<img width="637" alt="Screenshot 2023-05-05 at 5 08 10 pm" src="https://user-images.githubusercontent.com/124641980/236398757-9a60145b-3c35-4c38-9e66-f0cdd2cb8b43.png">

**Comment by fpezzola** [comment]
Hey @matias-gmg! 

Can you provide us the token address as well as the network you are facing this problem? 

thanks!

**Comment by matias-gmg** [comment]
The smart contract address is https://mumbai.polygonscan.com/address/0x26f09af814e12cd2fd6acf05bfa051bc33ab8b03 in Polygon Mumbai.

**Comment by julianariel** [comment]
> The smart contract address is https://mumbai.polygonscan.com/address/0x26f09af814e12cd2fd6acf05bfa051bc33ab8b03 in Polygon Mumbai.

hey @matias-gmg , thanks for the detail, this was helpful.

We've applied a new fix on #469 that will be pushed on the next release. In the meantime, you can try it out by building the latest version using the `master` branch.

cheers

**Comment by matias-gmg** [comment]
Thank you, I will wait for the next release to test.

**Comment by julianariel** [comment]
hey @matias-gmg ,

the fix is now ready on the latest [version 0.7.5](https://github.com/block-wallet/extension/releases/tag/v0.7.5) available in Chrome Store. 

thanks!
